bin/tx-prover/README.md

```diff
@@ -1,48 +1,60 @@
+meant to handle multiple workers with a queue for each one.
 
 ## Installation
 
-Install the prover binary for production using `cargo`:
+Install the prover worker and proxy binaries for production using `cargo`:
 
 ```sh
-cargo install miden-tx-prover --locked
+cargo install miden-tx-prover-worker --locked
+cargo install miden-tx-prover-proxy --locked
 ```
 
 This will install the latest official version of the prover. You can install a specific version using `--version <x.y.z>`:
 
 ```sh
-cargo install miden-tx-prover --locked --version x.y.z
+cargo install miden-tx-prover-worker --locked --version x.y.z
+cargo install miden-tx-prover-proxy --locked --version x.y.z
 ```
 
-You can also use `cargo` to compile the prover service from the source code if for some reason you need a specific git revision. Note that since these aren't official releases we cannot provide much support for any issues you run into, so consider this for advanced users only. The incantation is a little different as you'll be targetting this repo instead:
+To install both services at once from the source code using specific git revisions you can use `cargo install`. Note that since these aren't official releases we cannot provide much support for any issues you run into, so consider this for advanced users only:
 
 ```sh
 # Install from a specific branch
-cargo install --locked --git https://github.com/0xPolygonMiden/miden-base miden-tx-prover --branch <branch>
+cargo install --locked --git https://github.com/0xPolygonMiden/miden-base miden-tx-prover-worker --branch <branch>
+cargo install --locked --git https://github.com/0xPolygonMiden/miden-base miden-tx-prover-proxy --branch <branch>
 
 # Install a specific tag
-cargo install --locked --git https://github.com/0xPolygonMiden/miden-base miden-tx-prover --tag <tag>
+cargo install --locked --git https://github.com/0xPolygonMiden/miden-base miden-tx-prover-worker --tag <tag>
+cargo install --locked --git https://github.com/0xPolygonMiden/miden-base miden-tx-prover-proxy --tag <tag>
 
 # Install a specific git revision
-cargo install --locked --git https://github.com/0xPolygonMiden/miden-base miden-tx-prover --rev <git-sha>
+cargo install --locked --git https://github.com/0xPolygonMiden/miden-base miden-tx-prover-worker --rev <git-sha>
+cargo install --locked --git https://github.com/0xPolygonMiden/miden-base miden-tx-prover-proxy --rev <git-sha>
 ```
-
-If you want to build the prover from a local version, from the root of the workspace you can run:
+If you want to build from a local version, from the root of the workspace you can run:
 
 ```bash
-make install-prover
+make install-prover-worker
+make install-prover-proxy
 ```
 
-This step will also generate the necessary protobuf-related files.
+Note that for the prover worker you might need to add the `testing` feature in case you need. This step will also generate the necessary protobuf-related files.
 
-### Running the Service
+## Running the Service
```

Comment:
We should probably explain somewhere in this section what happens under various conditions. For example:

1. What happens if the worker crushes? Would it be automatically removed from the set of workers jobs get allocated to?
  a. Also, what happens if the worker reboots (i.e., goes down but then comes back up a short while later)?
3. What happens if the worker fails to prove something (i.e., returns an error)? Is the job assigned to a different worker or dropped entirely? Also, is the worker considered "faulty" in some way or does the load balancer continue assigning tasks to it?

Reply:
I added the answer of both questions in the README, let me know if it is not enough.

We could improve a lot the workers control from the proxy perspective when we have a health check endpoint in place, I just opened an issue to implement it.

Reply:
Also, added a simple retry method for requests in the case a worker is not working. It just retries N times (defined by a constant) with different workers, still using the round-robin approach.